uaclient/entitlements/__init__.py

```diff
@@ -38,12 +41,15 @@ def valid_services(allow_beta: bool = False) -> List[str]:
     allow_beta |= allow_beta_cfg
 
     if allow_beta:
-        return sorted(ENTITLEMENT_CLASS_BY_NAME.keys())
-
-    return sorted(
-        [
-            ent_name
-            for ent_name, ent_cls in ENTITLEMENT_CLASS_BY_NAME.items()
-            if not ent_cls.is_beta
-        ]
-    )
+        entitlements = ENTITLEMENT_CLASSES
+    else:
+        entitlements = [entitlement for entitlement in ENTITLEMENT_CLASSES if not entitlement.is_beta]
```

Comment:
I think we can remove the first if check:
```python
entitlements = ENTITLEMENT_CLASSES
if not allow_beta:
    entitlements = [entitlement for entitlement in entitlements if not entitlement.is_beta]
```

Reply:
This feels like stylistic changes, and also I think resetting a variable twice is a bit more costly than setting it once in different conditional paths. I get that it looks like less code to read, but timeit seems to think this suggestion is less desirable. Do folks have a strong opinion otherwise about this logical construct or whether one is more pythonic than the other?

```bash
>>> print(orig)
all_names = True
if all_names:
 a = 1
else:
 a = 2

>>> print(new)
all_names = True
a = 1
if all_names:
 a = 2

>>> timeit.timeit(stmt=orig, number=500000000)
11.050870887003839
>>> timeit.timeit(stmt=new, number=500000000)
13.152834953012643
```

Reply:
That's really interesting, I always thought that the cost of the conditional was heavier that setting up a variable. I see no problem keeping the version we have with this new evidence

Reply:
19 day old conversation :upside_down_face: but the performance of a branch is also dependent on how well it can be predicted by the cpu. The example scenario branch will always be guessed correctly so it will be fast. However, the performance difference is so minor that I don't think it matters here :smile:

Reply:
Premature optimization is evil or something, I don't remember